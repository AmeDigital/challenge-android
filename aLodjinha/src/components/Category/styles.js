import { StyleSheet } from 'react-native';

import material from '../../utils/MaterialDesign';

export default StyleSheet.create({
	image: {
		height: 100,
		width: 100
	},
	category: {
		justifyContent: 'center',
		alignItems: 'center',
		marginRight: 15,
		marginTop: 15,
		height: 120
	},
	description: {
		padding: 10,
		color: material.colorDark
	}
});
