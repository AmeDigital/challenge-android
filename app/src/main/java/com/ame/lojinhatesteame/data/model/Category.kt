package com.ame.lojinhatesteame.data.model

import com.google.gson.annotations.SerializedName
import java.io.Serializable

class Category : Serializable{
    @SerializedName("id") var id: Int = 0
    @SerializedName("urlImagem") var urlImage: String = ""
    @SerializedName("descricao") var description: String = ""
}