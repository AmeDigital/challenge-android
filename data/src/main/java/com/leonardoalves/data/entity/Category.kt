package com.leonardoalves.data.entity

import com.google.gson.annotations.SerializedName

data class Category(
	@SerializedName("id") val id: Int = 0,
	@SerializedName("urlImagem") val urlImagem: String? = null,
	@SerializedName("descricao") val descricao: String? = null
)