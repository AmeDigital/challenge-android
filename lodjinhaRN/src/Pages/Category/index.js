import React, { Component } from 'react';
import { FlatList, SafeAreaView} from 'react-native';

import ProductItem from '../../components/ProductItem';

export default class Category extends Component {

  state = {
    products: []
  }

  static navigationOptions = ({ navigation }) => ({
    title: navigation.state.params.category.descricao,
    headerTintColor: '#fff',
    headerStyle: {
      backgroundColor: '#5e2a84'
    }
  })

  async componentDidMount() {
    await this.getProducts()
  }

  _keyExtractor = (item, index) => String(item.id);

  getProducts = async () => {
    const response = await fetch(`https://alodjinha.herokuapp.com/produto/?categoriaId=${this.props.navigation.state.params.category.id}`);
    const products = await response.json();

    this.setState({
        products: products.data
    })
  }

  render() {
    return (
      <SafeAreaView>
        <FlatList
          data={this.state.products}
          keyExtractor={this._keyExtractor}
          renderItem={({ item }) => <ProductItem item={item} />} />
      </SafeAreaView>
    );
  }
}
