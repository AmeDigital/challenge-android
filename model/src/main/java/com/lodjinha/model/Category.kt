package com.lodjinha.model

import com.google.gson.annotations.SerializedName

data class Category(
    val id: Int,
    @SerializedName("descricao") val description: String,
    @SerializedName("urlImagem") val urlImage: String
)
