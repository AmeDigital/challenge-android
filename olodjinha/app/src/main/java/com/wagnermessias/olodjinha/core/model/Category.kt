package com.wagnermessias.olodjinha.core.model

import com.google.gson.annotations.SerializedName
import java.io.Serializable

class Category(
    @SerializedName("id")
    val id: Int,

    @SerializedName("urlImagem")
    val urlImage: String,

    @SerializedName("descricao")
    val description
    : String

): Serializable
