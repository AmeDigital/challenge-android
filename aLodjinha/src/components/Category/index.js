import React, { useState } from 'react';
import { Text, View, Image } from 'react-native';

import styles from './styles';

export default function Category({ item }) {
	const [ urlDefault, setUrlDefault ] = useState({ uri: item.urlImagem });

	function onError() {
		setUrlDefault(require('../../images/no-image.png'));
	}
	return (
		<View style={styles.category}>
			<Image style={styles.image} source={urlDefault} onError={onError} />
			<Text style={styles.description} numberOfLines={1}>
				{item.descricao}
			</Text>
		</View>
	);
}
