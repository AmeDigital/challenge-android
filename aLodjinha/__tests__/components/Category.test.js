import React from 'react';

import renderer from 'react-test-renderer';

import Category from '../../src/components/Category';

describe('Component Category', () => {
	test('should render without erros', () => {
		const tree = renderer.create(<Category item={{ urlImagem: 'urlimagem' }} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
