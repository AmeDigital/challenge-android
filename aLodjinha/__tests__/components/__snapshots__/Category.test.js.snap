// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Category should render without erros 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "height": 120,
      "justifyContent": "center",
      "marginRight": 15,
      "marginTop": 15,
    }
  }
>
  <Image
    onError={[Function]}
    source={
      Object {
        "uri": "urlimagem",
      }
    }
    style={
      Object {
        "height": 100,
        "width": 100,
      }
    }
  />
  <Text
    numberOfLines={1}
    style={
      Object {
        "color": "#2d3142",
        "padding": 10,
      }
    }
  />
</View>
`;
